codegen/type_converter.go

```diff
@@ -346,16 +346,30 @@ func (c *TypeConverter) genConverterForMap(
 		return err
 	}
 
+	valueStruct, isStruct := toFieldType.ValueSpec.(*compile.StructSpec)
+	sourceIdentifier := fromIdentifier
 	_, isStringKey := toFieldType.KeySpec.(*compile.StringSpec)
+
 	if !isStringKey {
-		return errors.Errorf(
-			"could not convert key (%s), map is not string-keyed.",
-			toField.Name,
+		_, isMapKey := toFieldType.KeySpec.(*compile.MapSpec)
+		if isMapKey {
```

Comment:
Sorry map key was an example. Doing `map<set<string>, string>` ; `map<list<string>, string` will also not work.

We probably need to check if the typedef resolves to `*compile.StringSpec` and reject every other type that's not string or typedef to string.

Reply:
basically just collections won't work right? Do you have an example how we can leverage compile.RootTypeSpec()?

Reply:
There should be examples of it. `git grep RootTypeSpec`